doc/rtd/reference/datasources.rst

```diff
@@ -0,0 +1,126 @@
+.. _datasources:
+
+Datasources
+***********
+
+Datasources are sources of configuration data for ``cloud-init`` that typically
+come from the user (i.e., user data) or come from the cloud that created the
```

Comment:
Not that I'm aware of! Like with a lot of modern grammatical rules, the rule is "it depends on what's most pleasing to the editor". Personally, I like it since people usually follow things like "for example", or "that is/that's to say" in conversation with a half-beat pause, so I tend to put a comma after e.g. and i.e. for that reason (although not always).